taichi/program/async_engine.cpp

```diff
@@ -414,30 +431,29 @@ bool AsyncEngine::fuse() {
 
       // replace all reference to the offloaded statement B to A
       irpass::replace_all_usages_with(task_a, task_b, task_a);
-      irpass::re_id(task_a);
 
       auto kernel = task_queue[i].kernel;
       irpass::full_simplify(task_a, /*after_lower_access=*/false, kernel);
-      task_queue[i].h = hash(task_a);
+      // For now, re_id is necessary for the hash to be correct.
+      irpass::re_id(task_a);
+
+      auto h = ir_bank_.get_hash(task_a);
+      task_queue[i].ir_handle = IRHandle(task_a, h);
+      ir_bank_.insert(std::move(cloned_task_a), h);
+      task_queue[i + 1].ir_handle = IRHandle(nullptr, 0);
+      ir_bank_.trash_bin.push_back(std::move(cloned_task_b));
```

Comment:
Not sure if we really need this, since `task_b`'s body has been fully moved into `task_a`... But maybe there could be unexpected references to `b`, could you add a TODO to see if we can remove this?